controller/instance_manager_controller.go

```diff
@@ -1726,8 +1732,19 @@ func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan) erro
 	return nil
 }
 
-func (imc *InstanceManagerController) canDeleteOrphanCR(imTerminating, autoDeleteEnabled, instanceExist, instanceCRScheduledBack bool) bool {
-	return imTerminating || autoDeleteEnabled || !instanceExist || instanceCRScheduledBack
+func (imc *InstanceManagerController) canDeleteOrphan(orphan *longhorn.Orphan, imTerminating, autoDeleteEnabled, instanceExist, instanceCRScheduledBack bool, autoDeleteGracePeriod int64) bool {
+	autoDeleteAllowed := false
+	if autoDeleteEnabled {
+		elapsedTime := time.Since(orphan.CreationTimestamp.Time)
+		if elapsedTime > gracePeriod {
+			autoDeleteAllowed = true
+		} else {
+			remainingTime := gracePeriod - elapsedTime
+			imc.logger.Debugf("Orphan %v is created %v seconds ago, will be deleted after %v seconds", orphan.Name, elapsedTime.Seconds(), remainingTime.Seconds())
```

Comment:
The log message could be misleading since all conditions are evaluated at the end of the function. Consider evaluating and logging the input conditions earlier for clarity.

Reply:
Updated. Thanks.